### DISCUSSION TITLE
get Documentation / information about how pyodide manage to package the Lib python directory

### DISCUSSION THREAD

**Discussion by julienfr112** [pyodide/pyodide#1454] [Q&A] [answered]
I'd like to know, for the how pyodide manage to package the file needed by python that reside in the Lib directory, and are necessary at runtime. 

A somehow related project, pyoxidizer, ( see https://pyoxidizer.readthedocs.io/en/v0.5.0/pyembed.html#default-initialization-of-python-importing-mechanism ) describe some solution with that problem.

How does Pyodide does to pacakge the Lib directory, as it seems required to run python in the browser ? I searched but didn't see informations about that in the doc, in the issuers or in the discussion. I also look at the patched applied to cpython, but didnt found anything related import in that.

**Comment by hoodmane** [comment, marked as answer]
We don't do anything fancy. We just use [--preload-file](https://emscripten.org/docs/porting/files/packaging_files.html#packaging-using-emcc) to put the entirety of `/usr/lib/python3.8` into the emscripten file system.
https://github.com/pyodide/pyodide/blob/master/Makefile#L48
Other than that, the normal loading system suffices. This is probably far from ideal in terms of speed / download size, but it works.

**Reply by julienfr112** [reply]
I didn't know about that : https://emscripten.org/docs/porting/files/file_systems_overview.html thanks again !
